src/rtFileDownloader.cpp

```diff
@@ -676,6 +724,412 @@ void rtFileDownloader::clearFileCache()
+                downloadRequest->setDownloadedData( invalidData, dataSize);
+                delete [] buffer;
+                fclose(fp);
+            }
+            mFileCacheMutex.unlock();
+            rtLogInfo("Reading from cache End for %s\n", downloadRequest->fileUrl().cString());
+        }
+      }
+    }
+    else
+#endif
+    {
+      nwDownloadSuccess = downloadByteRangeFromNetwork(downloadRequest);
+    }
+
+    if (!downloadRequest->executeCallback(downloadRequest->downloadStatusCode()))
+    {
+      if (mDefaultCallbackFunction != NULL)
+      {
+        (*mDefaultCallbackFunction)(downloadRequest);
+      }
+    }
+
+#ifdef ENABLE_HTTP_CACHE
+    // Store the network data in cache
+    if ((true == nwDownloadSuccess) &&
+        (true == downloadRequest->cacheEnabled())  &&
+        (downloadRequest->actualFileSize() == downloadRequest->downloadedDataSize()) &&
+        (downloadRequest->httpStatusCode() != 302) &&
+        (downloadRequest->httpStatusCode() != 307))
+    {
+      rtHttpCacheData downloadedData(downloadRequest->fileUrl(),
+                                     downloadRequest->headerData(),
+                                     downloadRequest->downloadedData(),
+                                     downloadRequest->downloadedDataSize());
+
+      if (downloadedData.isWritableToCache())
+      {
+        mFileCacheMutex.lock();
+        if (NULL == rtFileCache::instance())
+          rtLogWarn("cache data not added");
+        else
+        {
+          rtFileCache::instance()->addToCache(downloadedData);
+        }
+        mFileCacheMutex.unlock();
+      }
+    }
+
+    // Store the updated data in cache
+    if ((true == isDataInCache) && (cachedData.isUpdated()) && (downloadRequest->isDownloadOnly() == false))
+    {
+      rtString url;
+      cachedData.url(url);
+
+      mFileCacheMutex.lock();
+      if (NULL == rtFileCache::instance())
+          rtLogWarn("Adding url to cache failed (%s) due to in-process memory issues", url.cString());
+      rtFileCache::instance()->removeData(url);
+      mFileCacheMutex.unlock();
+      if (cachedData.isWritableToCache())
+      {
+        mFileCacheMutex.lock();
+        rtError err = rtFileCache::instance()->addToCache(cachedData);
+        if (RT_OK != err)
+          rtLogWarn("Adding url to cache failed (%s)", url.cString());
+
+        mFileCacheMutex.unlock();
+      }
+    }
+
+    if (true == isDataInCache)
+    {
+      downloadRequest->setHeaderData(NULL,0);
+      downloadRequest->setDownloadedData(NULL,0);
+    }
+#endif
+    clearFileDownloadRequest(downloadRequest);
+}
+
+bool rtFileDownloader::downloadByteRangeFromNetwork(rtFileDownloadRequest* downloadRequest)
+{
+    CURL *curl_handle = NULL;
+    CURLcode res = CURLE_OK;
+    char errorBuffer[CURL_ERROR_SIZE];
+    int multipleIntervals = 1;
+    size_t startRange = 0;
+    std::string byteRange("NULL");
```

Comment:
Please remove std::string usage.  Please use rtString instead.

Reply:
Changed to rtString and pushed again.